pkg/server/server.go

```diff
@@ -89,11 +106,11 @@ func newTCPListener(address string, port uint32, ch chan *net.TCPConn) (*tcpList
 				}).Warn("Failed to AcceptTCP")
 				return err
 			}
-			ch <- conn
+			ch <- conn.(*net.TCPConn)
```

Comment:
Is it safe to convert `conn` interface to net.TCPConn without validating, it will cause panic otherwise.

Reply:
Thank you for the review - I have changed it to the safe cast. While a type casting fail should be unreachable (network type `tcp4` or `tcp6` is given to Listen), it is certainly better to stay on the safe side as you suggested.

The PR should be ready, hence it is not marked as draft. However if there is anything missing (for example with respect to the added field into the proto file), please let me know.